src/tap/src/tapcell.cpp

```diff
@@ -231,10 +231,10 @@ int Tapcell::placeTapcells(odb::dbMaster* tapcell_master,
     x_loc = partially_overlap.right
                 ? partially_overlap.x_limit_right - tap_width
                 : x_loc;
-    if ((!overlap
-         || ((partially_overlap.left || partially_overlap.right)
-             && !isOverlapping(x_loc, tap_width, ori, row_insts)))
-        && x_loc + tap_width <= urx) {
+    bool partial_overlap = (partially_overlap.left || partially_overlap.right)
+                           && !isOverlapping(x_loc, tap_width, ori, row_insts);
+    bool in_rows = (x_loc + tap_width) <= urx;
+    if ((!overlap || partial_overlap) && in_rows) {
```

Comment:
Better though "!overlap || partial_overlap" is still a bit odd.  Can you be overlapped but not partially overlapped?

Reply:
Yes, you can. It is the case where a tapcell would be completely "inside" another cell. There is no geometry of the tapcell that is not outside of the other cell.